src/toil/batchSystems/kubernetes.py

```diff
@@ -881,6 +883,11 @@ def mount_host_path(volume_name: str, host_path: str, mount_path: str, create: b
                                                   name="runner-container",
                                                   resources=resources,
                                                   volume_mounts=mounts)
+
+        # In case security context rules are not allowed to be set
+        if self.config.kubernetes_privileged:
```

Comment:
This comment is kind of backwards from the if. We're going to do the thing the if controls if the user *does* want to do this thing, not if the thing isn't allowed to be done.
```suggestion
        # In case security context rules are not allowed to be set, we only apply
        # a security context at all if we need to turn on privileged mode.
        if self.config.kubernetes_privileged:
```